### PR TITLE
Make key vault settings optional for LLS in order to use CSS.

### PR DESCRIPTION
**Pull request by SergeyTeplyakov**
This PR makes key vault settings optional in oder to use Central Secret Store for CASaaS.

### CODE REVIEW

File: Public/Src/Cache/DistributedCache.Host/Service/Internal/DistributedContentStoreFactory.cs [hunk 1]
```diff
@@ -334,32 +334,10 @@ private List<string> GetAzureStorageSecretNames(StringBuilder errorBuilder)
...
-                    return null;
-                }
-
-                keyVaultSettingsString = keyVaultSettingsStringResult.ConnectionString;
-            }
-
-            _logger.Debug($"KeyVaultSettingsString: <non-null secret>, "
-                + $"EventHubSecretName: {_distributedSettings.EventHubSecretName}, AzureStorageSecretName: {_distributedSettings.AzureStorageSecretName}, GlobalRedisSecretName: {_distributedSettings.GlobalRedisSecretName}, SecondaryGlobalRedisSecretName: {_distributedSettings.SecondaryGlobalRedisSecretName}.");
+            _logger.Debug(
+                $"EventHubSecretName: {_distributedSettings.EventHubSecretName}, AzureStorageSecretName: {_distributedSettings.AzureStorageSecretName}, GlobalRedisSecretName: {_distributedSettings.GlobalRedisSecretName}, SecondaryGlobalRedisSecretName: {_distributedSettings.SecondaryGlobalRedisSecretName}.");
```
Comment (kewilli) [inline Public/Src/Cache/DistributedCache.Host/Service/Internal/DistributedContentStoreFactory.cs]: >EventHubSecretName [](start = 18, length = 18)

`nameof` ?

Comment (SergeyTeplyakov) [reply Public/Src/Cache/DistributedCache.Host/Service/Internal/DistributedContentStoreFactory.cs]: Sounds good.

Comment (SergeyTeplyakov) [reply Public/Src/Cache/DistributedCache.Host/Service/Internal/DistributedContentStoreFactory.cs]: Sounds good.

---
In reply to: [281360407](https://github.com/Microsoft/BuildXL/pull/257#discussion_r281360407) [](ancestors = 281360407)

**Pull request merged**